jupyter-extension/jupyterlab_pachyderm/pps_client.py

```diff
@@ -0,0 +1,97 @@
+import os.path
+import json
+import subprocess
+import sys
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+from tornado.httpclient import AsyncHTTPClient
+
+from .log import get_logger
+
+
+class PPSClient:
+    """Client interface for the PPS extension backend."""
+
+    def __init__(self):
+        self.client = AsyncHTTPClient()
+
+    async def generate(self, path, config):
+        """Generates the pipeline spec from the Notebook file specified.
+
+        Args:
+            path: The path (within Jupyter) to the Notebook file.
+            config: The PPS configuration for the Notebook file.
+        """
+        get_logger().debug(f"path: {path} | body: {config}")
+        # script = await self.client.fetch(f"http://localhost:8888/nbconvert/python{path}")
+        script_name = os.path.basename(path)
+        pipeline_spec = create_pipeline_spec("test_pipeline", "python:3.10", dict(pfs=dict(repo="data")), script_name)
+        return json.dumps(pipeline_spec)
+
+    async def create(self, path, config):
+        """Creates the pipeline from the Notebook file specified.
+
+        Args:
+            path: The path (within Jupyter) to the Notebook file.
+            config: The PPS configuration for the Notebook file.
+        """
+        get_logger().debug(f"path: {path} | body: {config}")
+        input_spec = config.get("input")
+        pipeline_name = config.get("pipeline_name")
+
+        with NamedTemporaryFile() as temp_config:
```

Comment:
Default config, but we're expecting the freshest copy of the config to be passed from the frontend, note that this purposely does not use the new ability of SAME to read from the notebook directly, as we want to explicitly pass the config every time.